src/detthread.rs

```diff
@@ -152,13 +152,19 @@ impl From<DetThreadId> for DetIdSpawner {
 
 #[derive(Clone, Serialize, Deserialize, Eq, PartialEq, Hash)]
 pub struct DetThreadId {
-    thread_id: Vec<u32>,
+    thread_id: [u32; 10],
+    extra_nodes: Vec<u32>,
+    size: u8,
 }
 
 use std::fmt::Debug;
 impl Debug for DetThreadId {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
-        write!(f, "ThreadId{:?}", self.thread_id)
+        let mut vec = Vec::from(&self.thread_id[..self.size as usize]);
```

Comment:
To avoid this vec allocation for every use of the debug trait we should just get rid of the `extra_nodes` fields and  `&[u32; 10]` can just formatted trivially as it implements Debug.